search/ranker.cpp

```diff
@@ -329,20 +329,74 @@ void Ranker::Init(Params const & params, Geocoder::Params const & geocoderParams
   m_tentativeResults.clear();
 }
 
-Result Ranker::MakeResult(RankerResult const & r) const
+Result Ranker::MakeResult(RankerResult const & rankerResult, bool needAddress,
+                          bool needHighlighting) const
 {
-  Result res = GenerateFinalResult(r, true /* needAddress */);
-  HighlightResult(m_params.m_tokens, m_params.m_prefix, res);
-  if (ftypes::IsLocalityChecker::Instance().GetType(r.GetTypes()) == ftypes::NONE)
+  ReverseGeocoder::Address addr;
```

Comment:
```c++
class AddressGetter 
{
public:
  AddressGetter(ReverseGeocoder const & geocoder, m2::PointD const & center): m_geocoder(geocoder), m_center(center) {}

  ReverseGeocoder::Address const & GetAddress()
  {
    if (m_computed)
      return m_address;
    m_address = m_geocoder.GetNearbyAddress(m_center, m_address);
    m_computed = true;
    return m_address;
  }

private:
  ReverseGeocoder & m_geocoder;
  m2::PointD const m_center;
  ReverseGeocoder::Address m_address;
  bool m_computed = false;
};

Reply:
This will simplify somehow the function.

Reply:
I've simplified even more.